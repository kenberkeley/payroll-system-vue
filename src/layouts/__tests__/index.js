/* global it, expect */
import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import VueRouter from 'vue-router'
import MainLayout from '../main.vue'

const localVue = createLocalVue()
localVue.use(VueRouter)

it('Main Layout', () => {
  const content = '<p>Hello World</p>'
  const wrapper = mount(MainLayout, {
    localVue,
    slots: { default: content },
    stubs: { RouterLink: RouterLinkStub }
  })
  expect(wrapper.find('main').html()).toContain(content)
  expect(wrapper.element).toMatchSnapshot()
})
