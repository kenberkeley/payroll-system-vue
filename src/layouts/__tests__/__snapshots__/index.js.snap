// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Layout 1`] = `
<div
  class="-main-layout"
>
  <header
    aria-label="main navigation"
    class="navbar has-shadow is-block"
    role="navigation"
  >
    <div
      class="-gradient-line"
    />
     
    <div
      class="container"
    >
      <div
        class="navbar-brand"
      >
        <a
          class="navbar-item"
        >
          <svg
            aria-label="MYOB Home"
            data-name="logo"
            preserveAspectRatio="xMidYMin slice"
            viewBox="0 0 283.5 116.4"
            width="6.25em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              MYOB
            </title>
             
            <defs>
              <lineargradient
                gradientUnits="userSpaceOnUse"
                id="logo-gradient"
                x1="0"
                x2="283.4646"
                y1="58.2083"
                y2="58.2083"
              >
                <stop
                  offset="0.36"
                  stop-color="#6100A5"
                />
                 
                <stop
                  offset="0.4557"
                  stop-color="#7001A0"
                />
                 
                <stop
                  offset="0.6409"
                  stop-color="#980293"
                />
                 
                <stop
                  offset="0.8947"
                  stop-color="#D9057E"
                />
                 
                <stop
                  offset="0.97"
                  stop-color="#ED0677"
                />
              </lineargradient>
            </defs>
             
            <g
              fill="url(#logo-gradient)"
            >
              <path
                d="M46.8,34.2 c-4.4-5.3-11.1-8.6-19.4-8.6C10.8,25.5,0,37.3,0,52.7V88h5.1c3.5-0.2,10.3-1.9,10.5-12V52.4c0-6.9,4.8-11.6,11.8-11.6 c10.3,0,11.4,8.2,11.4,11.6V88h5.1c3.4-0.2,10.2-1.9,10.5-11.9V52.4c0-6.9,5-11.6,11.8-11.6c10.3,0,11.5,8.2,11.5,11.6V88h5.1 c3.5-0.2,10.4-1.9,10.5-12.2V52.7c0-16-10.1-27.2-27.1-27.2C58.3,25.5,51.4,28.9,46.8,34.2"
              />
               
              <path
                d="M138.3,33.8l-13.1,36.4l-14.3-37.7c-2.2-6-7.8-5.6-7.8-5.6H90.9l24.2,59.2c0-0.1,0-0.1-0.1-0.2 c0,0,0,0.1,0.1,0.2c0,0.1,0.1,0.2,0.1,0.2l-0.1-0.3c0.8,2.4,0.6,3.7-0.2,6.1l-0.5,1.5c-2.8,7-6.8,8.9-15.2,8.4v0.1l5.8,14.3 c10.8-0.2,19-3.8,24.4-17.3l29.1-72.2l0,0l-10.9,0c0,0,0,0,0,0C147.1,26.9,140.9,27.1,138.3,33.8"
              />
               
              <path
                d="M170.1,57.7 c0-10.5,6.7-17.9,15-17.9c8.3,0,15,7.3,15,17.9c0,10.3-6.7,17.4-15,17.4C176.8,75.1,170.1,68,170.1,57.7 M154.5,57.6 c0,18.2,13.7,31.8,30.6,31.8c16.9,0,30.7-13.6,30.7-31.8c0-18.2-13.8-32-30.7-32C168.2,25.5,154.5,39.4,154.5,57.6"
              />
               
              <path
                d="M236.6,57.7c0-9.7,7-17.5,15.6-17.5 c8.6,0,15.6,7.6,15.6,17.5c0,9.7-7,17.1-15.6,17.1C243.6,74.9,236.6,67.2,236.6,57.7 M233.4,0L233.4,0 c-10.6,0-12.3,7.2-12.5,10.6V88l7.5,0c6,0.1,7.4-3.7,7.5-5.9v-0.3v-0.3c4.7,5,11.2,7.8,18.6,7.8c16,0,29-13.7,29-31.7 c0-18.1-13-31.8-29-31.8c-7,0-13.2,2.6-17.9,7.1V0H233.4z"
              />
            </g>
          </svg>
        </a>
         
        <a
          aria-expanded="false"
          aria-label="menu"
          class="navbar-burger"
          data-target="headerNavMenu"
          role="button"
        >
          <span
            aria-hidden="true"
          />
           
          <span
            aria-hidden="true"
          />
           
          <span
            aria-hidden="true"
          />
        </a>
      </div>
       
      <div
        class="navbar-menu"
        id="headerNavMenu"
      >
        <div
          class="navbar-end"
        >
          <a
            class="navbar-item"
          >
            
          Logout
        
          </a>
        </div>
      </div>
    </div>
  </header>
   
  <main>
    <section
      class="container"
    >
      <p>
        Hello World
      </p>
    </section>
  </main>
   
  <footer
    aria-label="navigation"
    class="navbar has-shadow is-block"
    role="navigation"
  >
    <div
      class="container"
      id="footer-links"
    >
      <div
        class="navbar-brand"
      >
        <a
          class="navbar-item"
        >
          <svg
            aria-label="MYOB Home"
            data-name="logo"
            preserveAspectRatio="xMidYMin slice"
            viewBox="0 0 283.5 116.4"
            width="7em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              MYOB
            </title>
             
            <defs>
              <lineargradient
                gradientUnits="userSpaceOnUse"
                id="logo-gradient"
                x1="0"
                x2="283.4646"
                y1="58.2083"
                y2="58.2083"
              >
                <stop
                  offset="0.36"
                  stop-color="#6100A5"
                />
                 
                <stop
                  offset="0.4557"
                  stop-color="#7001A0"
                />
                 
                <stop
                  offset="0.6409"
                  stop-color="#980293"
                />
                 
                <stop
                  offset="0.8947"
                  stop-color="#D9057E"
                />
                 
                <stop
                  offset="0.97"
                  stop-color="#ED0677"
                />
              </lineargradient>
            </defs>
             
            <g
              fill="url(#logo-gradient)"
            >
              <path
                d="M46.8,34.2 c-4.4-5.3-11.1-8.6-19.4-8.6C10.8,25.5,0,37.3,0,52.7V88h5.1c3.5-0.2,10.3-1.9,10.5-12V52.4c0-6.9,4.8-11.6,11.8-11.6 c10.3,0,11.4,8.2,11.4,11.6V88h5.1c3.4-0.2,10.2-1.9,10.5-11.9V52.4c0-6.9,5-11.6,11.8-11.6c10.3,0,11.5,8.2,11.5,11.6V88h5.1 c3.5-0.2,10.4-1.9,10.5-12.2V52.7c0-16-10.1-27.2-27.1-27.2C58.3,25.5,51.4,28.9,46.8,34.2"
              />
               
              <path
                d="M138.3,33.8l-13.1,36.4l-14.3-37.7c-2.2-6-7.8-5.6-7.8-5.6H90.9l24.2,59.2c0-0.1,0-0.1-0.1-0.2 c0,0,0,0.1,0.1,0.2c0,0.1,0.1,0.2,0.1,0.2l-0.1-0.3c0.8,2.4,0.6,3.7-0.2,6.1l-0.5,1.5c-2.8,7-6.8,8.9-15.2,8.4v0.1l5.8,14.3 c10.8-0.2,19-3.8,24.4-17.3l29.1-72.2l0,0l-10.9,0c0,0,0,0,0,0C147.1,26.9,140.9,27.1,138.3,33.8"
              />
               
              <path
                d="M170.1,57.7 c0-10.5,6.7-17.9,15-17.9c8.3,0,15,7.3,15,17.9c0,10.3-6.7,17.4-15,17.4C176.8,75.1,170.1,68,170.1,57.7 M154.5,57.6 c0,18.2,13.7,31.8,30.6,31.8c16.9,0,30.7-13.6,30.7-31.8c0-18.2-13.8-32-30.7-32C168.2,25.5,154.5,39.4,154.5,57.6"
              />
               
              <path
                d="M236.6,57.7c0-9.7,7-17.5,15.6-17.5 c8.6,0,15.6,7.6,15.6,17.5c0,9.7-7,17.1-15.6,17.1C243.6,74.9,236.6,67.2,236.6,57.7 M233.4,0L233.4,0 c-10.6,0-12.3,7.2-12.5,10.6V88l7.5,0c6,0.1,7.4-3.7,7.5-5.9v-0.3v-0.3c4.7,5,11.2,7.8,18.6,7.8c16,0,29-13.7,29-31.7 c0-18.1-13-31.8-29-31.8c-7,0-13.2,2.6-17.9,7.1V0H233.4z"
              />
            </g>
          </svg>
        </a>
         
        <a
          aria-expanded="false"
          aria-label="menu"
          class="navbar-burger"
          data-target="footerNavMenu"
          role="button"
        >
          <span
            aria-hidden="true"
          />
           
          <span
            aria-hidden="true"
          />
           
          <span
            aria-hidden="true"
          />
        </a>
      </div>
       
      <div
        class="navbar-menu"
        id="footerNavMenu"
      >
        <div
          class="navbar-end"
        >
          <a
            class="navbar-item"
          >
            
          Link 1
        
          </a>
           
          <a
            class="navbar-item"
          >
            
          Link 2
        
          </a>
           
          <a
            class="navbar-item"
          >
            
          Link 3
        
          </a>
        </div>
      </div>
    </div>
     
    <div
      id="footer-note"
    >
      <div
        class="container is-clearfix"
      >
        <p
          class="is-pulled-left"
        >
          
        Excepteur sint occaecat
      
        </p>
         
        <p
          class="is-pulled-right"
        >
          
        Lorem ipsum dolor sit amet, consectetur adipiscing elit
      
        </p>
      </div>
    </div>
     
    <div
      class="-gradient-line"
    />
  </footer>
</div>
`;
